pkg/registry/apps/daemonset/storage/storage_test.go

```diff
@@ -190,11 +198,91 @@ func TestWatch(t *testing.T) {
 }
 
 func TestShortNames(t *testing.T) {
-	storage, _, server := newStorage(t)
+	storage, server := newStorage(t)
 	defer server.Terminate(t)
-	defer storage.Store.DestroyFunc()
+	defer storage.DaemonSet.DestroyFunc()
 	expected := []string{"ds"}
-	registrytest.AssertShortNames(t, storage, expected)
+	registrytest.AssertShortNames(t, storage.DaemonSet, expected)
 }
 
 // TODO TestUpdateStatus
+
+func TestScaleGet(t *testing.T) {
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.DaemonSet.Store.DestroyFunc()
+
+	name := "foo"
+
+	var sts apps.DaemonSet
+	ctx := genericapirequest.WithNamespace(genericapirequest.NewContext(), metav1.NamespaceDefault)
+	key := "/daemonsets/" + metav1.NamespaceDefault + "/" + name
+	if err := storage.DaemonSet.Storage.Create(ctx, key, validDaemonSet, &sts, 0, false); err != nil {
+		t.Fatalf("error setting new daemonset (key: %s) %v: %v", key, validDaemonSet, err)
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(validDaemonSet.Spec.Selector)
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := &autoscaling.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              name,
+			Namespace:         metav1.NamespaceDefault,
+			UID:               sts.UID,
+			ResourceVersion:   sts.ResourceVersion,
+			CreationTimestamp: sts.CreationTimestamp,
+		},
+		Spec: autoscaling.ScaleSpec{
+			Replicas: validDaemonSet.Status.DesiredNumberScheduled,
+		},
+		Status: autoscaling.ScaleStatus{
+			Replicas: validDaemonSet.Status.DesiredNumberScheduled,
```

Comment:
IMO it would be nicer to expect other values than 0s here

Reply:
Good point, I defined non-zero values.